File: 实战前技术点/7.VideoToolBox-视频采集/VideoToolBox/ViewController.m
//
//  ViewController.m
//  VideoToolBox
//
//  Created by 韩俊强 on 2017/7/24.
//  Copyright © 2017年 HaRi. All rights reserved.
//

#import "ViewController.h"
#import <VideoToolbox/VideoToolbox.h>
#import <AVFoundation/AVFoundation.h>
#import "VideoCapture.h"

@interface ViewController ()<AVCaptureVideoDataOutputSampleBufferDelegate>

/** 视频捕捉对象 **/
@property (nonatomic, strong) VideoCapture *videoCapture;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
}

- (IBAction)startCaptureAction:(UIButton *)sender
{
    if (!TARGET_IPHONE_SIMULATOR) {
        [self.videoCapture startCapture:self.view];
    } else {
        NSLog(@"请使用真机测试, 模拟器不支持!");
    }
}

- (IBAction)stopCaptureAction:(UIButton *)sender
{
    if (!TARGET_IPHONE_SIMULATOR) {
        [self.videoCapture stopCapture];
    } else {
        NSLog(@"请使用真机测试, 模拟器不支持!");
    }
}

- (VideoCapture *)videoCapture
{
    if (!_videoCapture) {
        _videoCapture = [[VideoCapture alloc] init];
    }
    return _videoCapture;
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: 基础语法补充/闭包/基础语法补充/ViewController.swift
//
//  ViewController.swift
//  基础语法补充
//
//  Created by 韩俊强 on 2017/7/13.
//  Copyright © 2017年 HaRi. All rights reserved.
// 
//  闭包补充知识点

import UIKit

class ViewController: UIViewController {
    
    var httpTool : HttpTool?
    

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
        httpTool = HttpTool()
    }

    
    override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
        
        httpTool?.loadData(finished: { (jsonData) in
            print("在控制器中拿到数据:\(jsonData)")
        })
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

